### PR TITLE
Add support for running custom post processing script

### PR DESCRIPTION
**Pull request by batinicaz**
I've bee running a custom script to cleanup directories of deleted videos periodically but figured it'd be useful to support running such a script automatically after each batch of downloads has finished.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,6 +30,7 @@ COPY root/ /
 RUN set -x && \
     apt update && \
     apt install -y \
+        file \
```
Comment (Jeeaaasus) [inline Dockerfile:33]: Why is this necessary?

File: README.md [hunk 2]
```diff
@@ -89,6 +89,7 @@ Then configure the channels as explained in the [Configure youtube-dl](https://g
 | `youtubedl_watchlater` | `true` (`false`) | If you want to download your Watch Later playlist. Authentication is required.
 | `youtubedl_interval` | `1h` (`3h`) `12h` `3d` | If you want to change the default download interval. This can be any argument compatible with [gnu sleep](https://github.com/tldr-pages/tldr/blob/main/pages/linux/sleep.md).  Be mindful that if you download too frequently, you risk getting ip-banned by YouTube.<br />1 hour, (3 hours), 12 hours, 3 days.
 | `youtubedl_quality` | `720` (`1080`) `1440` `2160` | If you want to change the default download resolution.<br>720p, (1080p), 1440p, 4k.
+| `post_processing_script` | `/scripts/cleanup.bash` (`''`) | If you want to run a script after downloading has completed set this to the path to the script you want to run. Script must be available in the container.
```
Comment (Jeeaaasus) [inline README.md:92]: I would like you to remove this.

We should simply make it discover and run both post & pre-execution scripts if the files exist, no need to make the name user definable.

Do you have any issues with me putting the scripts in /config/?

File: root/app/youtube-dl/youtube-dl.sh [hunk 3]
```diff
@@ -50,6 +50,15 @@ else
...
 
+if [ -f "$post_processing_script" ]
+then
+  echo "[post processing] Running custom post processing script: $post_processing_script"
+  "$post_processing_script"
+  echo "[post processing] Post processing completed, refer to above output for additional info"
+else
+  echo "[post processing] post_processing_script not set, skipping"
+fi
+
```
Comment (Jeeaaasus) [inline root/app/youtube-dl/youtube-dl.sh:61]: replace with:
```

if [ -f '/config/post-execution.sh' ]
then
  echo '[post-execution] running post-execution script...'
  bash '/config/post-execution.sh'
  echo '[post-execution] finished post-execution script.'
fi

echo "$youtubedl_binary version: $youtubedl_version"

```
And then on line 19 also add:
```

if [ -f '/config/pre-execution.sh' ]
then
  echo '[pre-execution] running pre-execution script...'
  bash '/config/pre-execution.sh'
  echo '[pre-execution] finished pre-execution script.'
fi

```

**Comment by Jeeaaasus** [comment]
Hello @batinicaz :)

I like the idea, sounds useful.
I have however left a couple of in-line comments.

After those changes I will update README to mention the feature.

File: Dockerfile (hunk 1 above)
Comment (batinicaz) [reply Dockerfile:33]: I'm using it in my post execution script so I can find directories which don't contain video files (so only left over metadata from yt-dlp). 

Happy to remove if needed but given it's small package seemed a nice to have with the introduction of post processing scripts that might want to do similar things :-)

File: README.md (hunk 2 above)
Comment (batinicaz) [reply README.md]: That makes sense, keeping them in the already mounted directory seems reasonable so no additional mounting is needed

**Comment by batinicaz** [comment]
I think this addresses the comments - https://github.com/Jeeaaasus/youtube-dl/compare/e391c00905cb662f5167c9d738c7560135a721f6..795999da4d4abe854ea28a6b908d87d022f14615

Let me know if you think further adjustments are needed and I'm happy to address them

File: Dockerfile (hunk 1 above)
Comment (Jeeaaasus) [reply Dockerfile:33]: Okey, let's keep it.

File: root/app/youtube-dl/youtube-dl.sh [hunk 4]
```diff
@@ -17,6 +17,14 @@ if $youtubedl_watchlater; then echo ":ytwatchlater | --playlist-end '-1' --no-pl
 if ! $youtubedl_args_format; then exec+=" --format '$(cat '/config.default/format')'"; fi
 if ! $youtubedl_args_download_archive; then exec+=" --download-archive '/config/archive.txt'"; fi
 
+
```
Comment (Jeeaaasus) [inline root/app/youtube-dl/youtube-dl.sh:20]: one too many empty lines here.

Comment (batinicaz) [reply root/app/youtube-dl/youtube-dl.sh]: Fixed - https://github.com/Jeeaaasus/youtube-dl/compare/795999da4d4abe854ea28a6b908d87d022f14615..6a9e3c849840ab54cc81047eff61f74ea3d155ea

**Comment by Jeeaaasus** [comment]
Thank you for caring enough to suggest improvements to my little hobby project!

Keep in mind that the changes will be built into the image `unstable` automatically but will only be included in the image `latest` along with the next yt-dlp release version.
